src/monster.cpp

```diff
@@ -1332,19 +1331,49 @@ int monster::dodge_roll()
     return dice(numdice, 10);
 }
 
-int monster::fall_damage() const
+float monster::fall_damage_mod() const
 {
- if (has_flag(MF_FLIES))
-  return 0;
- switch (type->size) {
-  case MS_TINY:   return rng(0, 4);  break;
-  case MS_SMALL:  return rng(0, 6);  break;
-  case MS_MEDIUM: return dice(2, 4); break;
-  case MS_LARGE:  return dice(2, 6); break;
-  case MS_HUGE:   return dice(3, 5); break;
- }
+    if( has_flag(MF_FLIES) ) {
+        return 0.0f;
+    }
+
+    switch (type->size) {
+        case MS_TINY:
+            return 0.2f;
+        case MS_SMALL:
+            return 0.6f;
+        case MS_MEDIUM:
+            return 1.0f;
+        case MS_LARGE:
+            return 1.4f;
+        case MS_HUGE:
+            return 2.0f;
+        default:
```

Comment:
Through this is more a matter of style, but a switch over an enum values that handles *all* the possible values should not contain a default case. In this case the `return 0;` can be moved behind the switch.

Not having a default there allows the compiler to issue a warning when some enumeration values are not handled in the switch.